### DISCUSSION TITLE
Welcome to undangan-api Discussions!

### DISCUSSION THREAD

**Discussion by dewanakl** [dewanakl/undangan-api#11] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by drsydn** [comment]
mas, izin tanya.. saya kok gabisa login ke dashboard yaa? padahal udah ubah semua nya .. termasuk data-url di dashboard

apa yang salah yaa?
![Cuplikan layar 2025-05-01 135348](https://github.com/user-attachments/assets/fbb914e7-257e-4e9b-aae8-fd9026a85757)

**Reply by dewanakl** [reply]
sebelumnya sudah migrasi belum mas? coba check di tabel user apakah ada datanya?

**Reply by drsydn** [reply]
sudah mas, saya pake postgresql heroku sih, soalnya elephant ga bisa ..

btw postman nya harus di setting juga ga yaa? saya awam mas, mon maap

**Reply by drsydn** [reply]
pas deploy ke vercel kaya gini mas


npm error code ENOENT
npm error syscall open
npm error path /vercel/path0/package.json
npm error errno -2
npm error enoent Could not read package.json: Error: ENOENT: no such file or directory, open '/vercel/path0/package.json'
npm error enoent This is related to npm not being able to find a file.
npm error enoent
npm error A complete log of this run can be found in: /vercel/.npm/_logs/2025-05-01T07_44_07_461Z-debug-0.log
Error: Command "npm install" exited with 254

**Reply by dewanakl** [reply]
Untuk postman optional mas, jika pingin tau seluruh api ada di postmannya.
coba check file package.json apakah ada mas?

**Comment by drsydn** [comment]
kalo di repo undangan nya sih ada.. kalo di api gaada mas

**Reply by drsydn** [reply]
sudah bisa mas,, tadi tak otak atik wkwk..

makasih yaa mas

**Reply by dewanakl** [reply]
sama sama mas

**Reply by cobabisa002-arch** [reply]
mas pake perintah npm run build:public folder public nya tdk mncul ,hanya folder dist saja...kira2 kenapa ya folder public ny tdk mncul

**Reply by dewanakl** [reply]
mungkin permissionnya mas, coba pake sudo kalau di linux atau run as admin kalau di windows

**Comment by hd-wi** [comment]


**Reply by dewanakl** [reply]
Iya betul, bisa mas

**Comment by dheoprasetyo** [comment]
ijin tanya mas kenapa setelah di run api nya return nya

{
    "id": "18455e0c-307f-4554-bb22-fb2eb37b6ed6",
    "data": null,
    "error": [
        "user_agent panjang minimal: 64"
    ]
}

ini semua ya ? Login pun sama

**Reply by dewanakl** [reply]
sebelumnya masnya coba hit memakai postman? jika iya bisa set header user agent dummy yang panjangnya 64 mas

**Reply by dheoprasetyo** [reply]
iya via postman, maaf maksudnya gimana ya mas ? Soalnya di dokumentasinya engga ada

**Reply by dewanakl** [reply]
<img width="1314" height="218" alt="image" src="https://github.com/user-attachments/assets/73a2871d-34a7-431d-9238-efc6e50bfd1c" />

Betul mas, di dokumentasi belum tersedia, bisa coba isi dengan string random saja mas

**Reply by dheoprasetyo** [reply]
thx mas

**Reply by lajujaya95-art** [reply]
tutor cara migrasi database undangan-api  ke localhost di xampp kek gmn caranya mas

**Reply by dewanakl** [reply]
bisa lihat pada discussion ini mas:
https://github.com/dewanakl/undangan/discussions/88

**Comment by cobabisa002-arch** [comment]
Mas kalo mau buat undangan nya di add-on domain semisaal domain nya undangan.com
Nah file undangan-api nya di upload dmn dan publik nya apa di jadikan satu juga apa buat folder baru.

**Reply by dewanakl** [reply]
bisa bikin subdomain mas

**Reply by cobabisa002-arch** [reply]
Sudah. Bikin subdomain.
Yg bingung folder undangan-api nya di taro dmn nanti mas...dan folder publik yg ad di dalam undangan nya di move kmn mas

**Reply by dewanakl** [reply]
owh, di undangan api dalamnya ada folder public nah folder public ini yang nanti pada subdomainnya
isi undangan-api seperti vendor dll, bebas taruh dimana mas, terpenting nanti edit file index.php pada public subdomain untuk sesuaikan path nya saja dengan folder vendornya

**Reply by dewanakl** [reply]
untuk folder sanboxfile sebaiknya jangan di folder public nya karena nanti bisa di akses, dan folder vendor ada di dalam sanboxfile, tinggal ubah saja pada index.php untuk sesuaikan path dari folder sanboxfile

**Reply by cobabisa002-arch** [reply]
sudah saya pindah kan folder sanboxfile nya mas,,tapi saya cari index.php di vendor folder sanboxfile tdk ada,,,index.php adanya di folder sandbox saja mas
<img width="857" height="414" alt="Capture" src="https://github.com/user-attachments/assets/dbc441b8-0876-4e8f-8b0c-6c7cc6ac9671" />

**Reply by dewanakl** [reply]
index.php betul ada di folder public (yang mengarah ke domain) atau dalam kasus masnya itu namanya sanbox

lalu bisa edit file index.php lalu ubah path yang mengarah ke folder undangan-api (berisi vendor, app, cache dll) atau dalam kasus ini namanya sanboxfile

*untuk nama tidak harus sama seperti di video mas, bisa bebas, misal sanboxfile jadi undangan-api dll

**Comment by cobabisa002-arch** [comment]
saya mau export database yg ad si postgres untuk saya upload di hosting ,tapi eror mas ketika jalankan printah php saya migarasi:dump

<img width="967" height="408" alt="1" src="https://github.com/user-attachments/assets/f36ae257-9ac4-4c13-a933-b3742279f9da" />

**Reply by dewanakl** [reply]
pastikan databasenya nyala juga mas

**Comment by lajujaya95-art** [comment]
klo muncul ini ketika web dbuka msalah di mana mas
This page isn’t working
sandbox.lancarjaya.com is currently unable to handle this request.
HTTP ERROR 500

**Reply by dewanakl** [reply]
Bisa coba enable debugnya dulu aja mas pada hostingnya

**Reply by cobabisa002-arch** [reply]
di file env ya,,,ubah ke true bkn

**Reply by dewanakl** [reply]
iya betul, sama di hostingnya juga, cek dibagian php gitu mas

**Comment by lajujaya95-art** [comment]
saat import database muncul ini mas,,sql ini saya dapatkan dari php saya migrasi:dump
kira2 salah di mana ya mas

<img width="1295" height="390" alt="Capture" src="https://github.com/user-attachments/assets/f19f1cd5-f1a4-4e30-bf54-3b86171097a9" />

**Reply by dewanakl** [reply]
apakah sebelumnya memakai database yang sama mas?

**Reply by lajujaya95-art** [reply]
ia betul mas,,ini database yg sama..dari  undangan-api..

**Reply by dewanakl** [reply]
Sorry, sama disini sebelumnya pake postgresql untuk generate sql dan import memakai mysql, apakah begitu mas?

**Reply by lajujaya95-art** [reply]
ia betul mas,
oia mas klo ini saya ikuti cara di yutub mas cara git init selalu eror pdahal sudah sesuai perintah yg mas contohkan di yutub
<img width="954" height="490" alt="Capture" src="https://github.com/user-attachments/assets/31a18eb4-49fc-42d2-9fc0-cc5244696f48" />

**Reply by dewanakl** [reply]
karena berbeda database jadi berbeda juga sql nya mas, untuk git bisa config dulu mas

**Reply by lajujaya95-art** [reply]
> Sorry, sama disini sebelumnya pake postgresql untuk generate sql dan import memakai mysql, apakah begitu mas?

ia betul mas,,cmn waktu sudah berhasil di generete ke mysql ,pas di import selalu gagal

**Reply by cobabisa002-arch** [reply]
udah bsa mas,,oia kalo deploy di https://www.netlify.com/ bsa g mas kira2

**Reply by cobabisa002-arch** [reply]
<img width="936" height="615" alt="Capture" src="https://github.com/user-attachments/assets/ec4ed8be-9420-4e7d-91b3-f91f9ea85d1d" />
klo sprti ini sudah berhasil ke hub y mas

**Reply by dewanakl** [reply]
iya sudah bisa ke github
bisa ke vercel mas, nanti pake supabase

**Comment by cobabisa002-arch** [comment]
sudah bisa mas mksih,oia untuk login dasboard di local pake user yg ada di panduan yg mas buat ya

<img width="1261" height="600" alt="Capture" src="https://github.com/user-attachments/assets/293d4f92-9446-4d72-a347-b985100302e0" />

**Reply by lajujaya95-art** [reply]
klo di login muncul ini eror ny dmn mas
<img width="888" height="585" alt="Capture" src="https://github.com/user-attachments/assets/ad0c8e6a-8bfc-46d9-be82-eea704adca73" />

**Reply by dewanakl** [reply]
pastikan di database sudah ada mas, user@example nya

**Reply by lajujaya95-art** [reply]
ini isi database kan mas, sudah saya export ke mysql via dbeaver tapi ,,saat import ke phpmyadmin selalu gagal

<img width="788" height="444" alt="1" src="https://github.com/user-attachments/assets/4d5975ee-62a3-4e00-bcbd-de580452cdb6" />
<img width="1066" height="539" alt="fg" src="https://github.com/user-attachments/assets/becfd5b5-944b-4ca5-85b1-008dba8cb1fd" />

**Reply by dewanakl** [reply]
betul itu file yang masih dalam kode php.
kalau dari postgresql ke mysql itu tidak bisa mas, sudah beda syntax
seharusnya ketika export memakai mysql, importnya memakai mysql juga mas

**Reply by lajujaya95-art** [reply]
ia mas itu file database dalam kode php saya export ke msyql ,tapi saat impor gagal padahal sudah format sql,
oia mas pertanyaan saya di pgadmin tdk bsa import ya hanya export saja,,,saya coba koneksikan program postgres ke server hosting nya gagal trus hehe,thanks mas atas jawaban dari awal ya
<img width="1233" height="296" alt="Capture" src="https://github.com/user-attachments/assets/e3b918ce-692f-4ff8-aa47-a36ccae78519" />

**Reply by dewanakl** [reply]
Sekarang sudah ada endpoint buat migration mas
jadi tidak perlu repot repot lagi mikirin database

tinggal buka saja endpoint "/migrate?hash=xyz"
xzy diganti dengan sha3-512 dari APP_KEY pada env

**Comment by lajujaya95-art** [comment]
KLO keterangan email TDK valid knp y mas...

Pada Min, 5 Okt 2025 21.23, Dewana Kretarta Lokeswara <
***@***.***> menulis:

**Reply by dewanakl** [reply]
sudah sesuai sama database mas?
